### PR TITLE
Replace stackName from URL parameters with hostURL

### PR DESCRIPTION
**Pull request by luiscuenca**
This PR removes all logic that extract the **webSocket** address from the `stackName` URL param. Now this information is passed using a `hostURL` function parameter.

### CODE REVIEW

File: src/classes/HiFiCommunicator.ts [hunk 1]
```diff
@@ -187,24 +187,14 @@ export class HiFiCommunicator {
...
-     * it is very likely that we are running a test in a browser. Tests running in a browser from that hostname should assume that the WebRTC Signaling Address
-     * is at the same host from which the test is served.
-     * 3. If the code is running in the browser context, and our code compilation processes have specified the build mode as "production", we should use the production
-     * WebRTC signaling connection address.
-     * 4. If a developer has passed a `stackName` parameter into this `connectToHiFiAudioAPIServer()` call, use a WebRTC signaling address based on that `stackName`.
-     *     - If the passed `stackName` is a URL, that URL will be used as the WebRTC Signaling Address directly.
-     * 5. If the code is running in the NodeJS context, we will use the "production" `stackName`, and use a WebRTC signaling address based on that `stackName`.
-     * 6. If none of the above logic applies, we will use the default "staging" WebRTC signaling connection address.
+     * @param hostURL An URL that will be used to create a valid WebRTC signaling address. `wws://${hostURL}:8001/?token=`
```
Comment (zfox23) [inline src/classes/HiFiCommunicator.ts:190]:
```suggestion
     * @param hostURL An URL that will be used to create a valid WebRTC signaling address. The passed `hostURL` will be used to construct a signaling address of the form: `wss://${hostURL}:8001/?token=`
```

File: src/classes/HiFiCommunicator.ts [hunk 2]
```diff
@@ -187,24 +187,14 @@ export class HiFiCommunicator {
...
+     * If the developer does not pass a hostURL parameter, a default URL will be used instead. HiFiConstants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT
```
Comment (zfox23) [inline src/classes/HiFiCommunicator.ts:191]:
```suggestion
     * If the developer does not pass a `hostURL` parameter, a default URL will be used instead: See: {@link DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT}
```

File: src/classes/HiFiCommunicator.ts [hunk 3]
```diff
@@ -187,24 +187,14 @@ export class HiFiCommunicator {
...
+     * Reading this parameter from the URL should be implemented by the developer as part of the application code.
```
Comment (zfox23) [inline src/classes/HiFiCommunicator.ts:192]: What do you mean by this?

File: src/classes/HiFiCommunicator.ts [hunk 4]
```diff
@@ -214,56 +204,17 @@ export class HiFiCommunicator {
...
-                }
-            } else if (!isBrowserContext) {
-                webRTCSignalingAddress = `${HiFiConstants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT}/?token=`;
-            } else {
-                webRTCSignalingAddress = webRTCSignalingAddress.replace('$STACKNAME', 'api-staging-02');
-            }
+            hostURLSafe = new URL(hostURL).hostname;
+        } catch(e) {
+            // If hostURL is not defined, we assign the default URL
+            hostURLSafe = hostURL ? hostURL : HiFiConstants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT;
```
Comment (zfox23) [inline src/classes/HiFiCommunicator.ts:213]: why wouldn't we want to do this:
```suggestion
            hostURLSafe = HiFi4Constants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT;
```
?

**Review by zfox23 — CHANGES_REQUESTED** [review]

File: src/classes/HiFiCommunicator.ts (hunk 3 above)
Comment (luiscuenca) [reply src/classes/HiFiCommunicator.ts:192]: This is a note that makes reference to this change. Basically we say that we are not getting information from the URL params anymore. Now we rely on what URL is passed during the connection call. This means that it is the application itself that needs to get this address from the URL params if necessary, so the developer is now in charge of implementing this.

File: src/classes/HiFiCommunicator.ts (hunk 4 above)
Comment (luiscuenca) [reply src/classes/HiFiCommunicator.ts:213]: The first thing we do is trying to catch a complete URL (i.e. `https://api.highfidelity.com/`). If that fails we assume that either `hostURL` is undefined or it is not a complete URL (i.e `api.highfidelity.com`). Keep in mind that `new URL("api.highfidelity.com")` will fail.

File: src/classes/HiFiCommunicator.ts (hunk 1 above)
Comment (luiscuenca) [reply src/classes/HiFiCommunicator.ts]: Fixed

File: src/classes/HiFiCommunicator.ts (hunk 2 above)
Comment (luiscuenca) [reply src/classes/HiFiCommunicator.ts]: Fixed

File: src/classes/HiFiCommunicator.ts (hunk 3 above)
Comment (zfox23) [reply src/classes/HiFiCommunicator.ts:192]: oh okay i see!

File: src/classes/HiFiCommunicator.ts (hunk 4 above)
Comment (zfox23) [reply src/classes/HiFiCommunicator.ts:213]: aha! okay, thanks for the explanation.

**Review by zfox23 — APPROVED** [review]

File: src/classes/HiFiCommunicator.ts (hunk 4 above)
Comment (hd23408) [reply src/classes/HiFiCommunicator.ts:213]: @luiscuenca Sorry, late to the party here -- does that mean that when a dev passes in "hostURL" it needs to be a complete URL of the format `https://whatever.highfidelity.com/` and NOT just `whatever.highfidelity.com`? I had interpreted from the "`wws://${hostURL}:8001/?token=`" comment you had above that `whatever.highfidelity.com` was the right format. Can you clarify in the comments exactly which format is expected? Thank you!!

**Pull request merged**